source/computer/linux101.rst

```diff
@@ -49,15 +49,317 @@ Linux 入门
+    # 移动 hello-world-cp.txt 和 hello-world-mv.txt 文件到 source 目录下
+    $ mv hello-world-cp.txt hello-world-mv.txt source
+    ls
+    destination  seiso-learn.txt  source
+    $ ls source
+    hello-world-cp.txt  hello-world.txt  hello-world-mv.txt  seiso-learn.txt
+
+    # 移动 source 目录为同一目录下的 source-mv 目录（相当于重命名）
+    $ mv source source-mv
+    ls
+    destination  seiso-learn.txt  source-mv
+    # 移动 source-mv 目录到同一目录下的 destination 目录下
+    $ mv source-mv destination
+    $ ls
+    destination  seiso-learn.txt
+    $ ls destination
+    hello-world.txt  seiso-learn.txt  source  source-mv
+
+删除文件和目录::
+
+    # 使用 rm（remove，即删除）命令删除 seiso-learn.txt 文件
+    $ rm seiso-learn.txt
+    $ ls
+    destination
+    # 删除 destination 目录
+    $ rm -r destination
+    $ ls
+
+    # 新建 tmp 目录
+    $ mkdir tmp
+    $ ls
+    tmp
+    # 使用 rmdir（remove directory，即删除目录）命令删除空目录 tmp。目录不为空时则不删除
+    $ rmdir tmp
+    $ ls
+
+.. warning::
+
+   使用 ``rm`` 命令时一定要小心再小心，不要误删重要文件。可以先把要删除的文件移动到
+   某目录下（如 :file:`~/trash`\ ），再删除。
+
+建立文件和目录的链接::
+
+    # 新建 hello-world.txt 文件和 source 目录
+    $ touch hello-world.txt
+    $ mkdir source
+    $ ls
+    hello-world.txt  source
+
+    # 使用 ln（link，即链接）命令建立 hello-world.txt 的硬链接 hello-world-hard.txt
+    $ ln hello-world.txt hello-world-hard.txt
+    $ ls
+    hello-world-hard.txt  hello-world.txt  source
+    # 使用 ls 命令的 -l 选项可以查看文件和目录的详细信息
+    $ ls -l
+    total 0
+    -rw-r--r-- 2 seismo-learn seismo-learn 0 Feb  8 14:55 hello-world-hard.txt
+    -rw-r--r-- 2 seismo-learn seismo-learn 0 Feb  8 14:55 hello-world.txt
+    drwxr-xr-x 2 seismo-learn seismo-learn 6 Feb  8 14:55 source
+
+    # 建立 hello-world.txt 的软链接（也叫符号链接）hello-world-soft.txt
+    $ ln -s hello-world.txt hello-world-soft.txt
+    $ ls -l
+    total 0
+    -rw-r--r-- 2 seismo-learn seismo-learn  0 Feb  8 14:55 hello-world-hard.txt
+    -rw-r--r-- 2 seismo-learn seismo-learn  0 Feb  8 14:55 hello-world.txt
+    lrwxrwxrwx 1 seismo-learn seismo-learn 14 Feb  8 14:57 hello-world-soft.txt -> hello-world.txt
+    drwxr-xr-x 2 seismo-learn seismo-learn  6 Feb  8 14:55 source
+
+    # 建立 source 目录的软链接
+    $ ln -s source source-soft
+    $ ls -l
+    total 0
+    -rw-r--r-- 2 seismo-learn seismo-learn  0 Feb  8 14:55 hello-world-hard.txt
+    -rw-r--r-- 2 seismo-learn seismo-learn  0 Feb  8 14:55 hello-world.txt
+    lrwxrwxrwx 1 seismo-learn seismo-learn 14 Feb  8 14:57 hello-world-soft.txt -> hello-world.txt
+    drwxr-xr-x 2 seismo-learn seismo-learn  6 Feb  8 14:55 source
+    lrwxrwxrwx 1 seismo-learn seismo-learn  6 Feb  8 14:58 source-soft -> source
+
+    # 删除 source 目录的软链接
+    $ rm source-soft
+    $ ls
+    hello-world-hard.txt  hello-world.txt  hello-world-soft.txt  source
+    # 删除 hello-world.txt 文件的软链接
+    $ rm hello-world-soft.txt
+    $ ls
+    hello-world-hard.txt  hello-world.txt  source
+    # 删除 hello-world.txt 文件的硬链接
+    $ rm hello-world-hard.txt
+    $ ls
+    hello-world.txt  source
```

Comment:
Is this part necessary?

Reply:
New babies may have to how to remove the link, but I simplified it in dbffffc.